docs: give better guidance about how to configure the agent TLS CA (#10718)
hashicorp/consul: website/content/docs/agent/options.mdx

TLS is used by the HTTP API, server RPC, and xDS interfaces. Some of these settings may also be
applied automatically by [auto_config](#auto_config) or [auto_encrypt](#auto_encrypt).

> ~> **Security Note:** The Certificate Authority (CA) specified by `ca_file` and `ca_path`
> should use a private CA, not a public one.  We also recommend using a separate CA for
> Consul and not sharing the CA with any other systems. Any certificate signed by the
> CA will be allowed to communicate with the cluster and a specially crafted certificate
> signed by the CA can gain full read and write access to Consul.

Reviewer: This is a great addition to the docs!

Small wording suggestion, what do you think?

```suggestion
~> **Security Note:** The Certificate Authority (CA) specified by `ca_file` or `ca_path`
should be a private CA, not a public one. We recommend using a dedicated CA
which should not be used with any other systems. Any certificate signed by the
CA will be allowed to communicate with the cluster and a specially crafted certificate
signed by the CA can be used to gain full access to Consul.
```

Author: Sounds great, thank you!
